### ISSUE TITLE
pypy's fail to build from source with GCC 14 due to enforcing -Wincompatible-pointer-types

### ISSUE THREAD

**Issue by stratakis** [pypy/pypy#4926]
Possibly a duplicate or leftovers from #4041

On Fedora, the latest development versions contain GCC 14, which is stricter and enforces -Wincompatible-pointer-types. pypy2.7, pypy3.9 and pypy3.10 all currently fail. More specifically:

On pypy2.7:
```
_gdbm_cffi.c: In function ‘_cffi_d_gdbm_open’:
_gdbm_cffi.c:965:36: error: passing argument 5 of ‘gdbm_open’ from incompatible pointer type [-Wincompatible-pointer-types]
  965 |   return gdbm_open(x0, x1, x2, x3, x4);
      |                                    ^~
      |                                    |
      |                                    void (*)(void)
In file included from _gdbm_cffi.c:571:
/usr/include/gdbm.h:127:29: note: expected ‘void (*)(const char *)’ but argument is of type ‘void (*)(void)’
  127 |                             void (*)(const char *));
      |                             ^~~~~~~~~~~~~~~~~~~~~~
```

And on pypy3.9 and 3.10:
```
_tkinter/tklib_cffi.c: In function ‘_cffi_d_Tcl_Merge’:
_tkinter/tklib_cffi.c:2734:24: error: passing argument 2 of ‘Tcl_Merge’ from incompatible pointer type [-Wincompatible-pointer-types]
 2734 |   return Tcl_Merge(x0, x1);
      |                        ^~
      |                        |
      |                        char **
In file included from /usr/include/tcl.h:2421,
                 from _tkinter/tklib_cffi.c:572:
/usr/include/tclDecls.h:584:66: note: expected ‘const char * const*’ but argument is of type ‘char **’
  584 | EXTERN char *           Tcl_Merge(int argc, CONST84 char *const *argv);
```

The issues can be resolved by modifying the pointer to what's expected.

**Comment by mattip** [comment]
Thanks. Fixed in 8831ebf1cd and 789f964ff.

**Comment by stratakis** [comment]
Thanks for the fixes!
